testcase/unittest/test_tx_message.py

```diff
@@ -0,0 +1,166 @@
+
+        # then
+        assert queue.max_tx_size == given_max_tx_size
+        assert queue.max_tx_count == given_max_tx_count
+
+    @pytest.mark.parametrize("tx_items", [tx_count], indirect=True)
+    def test_tx_messages_queue(self, tx_items: List[Tuple[Transaction, TransactionVersioner]]):
+        print()
+        # given add 50 tx data to TxMessagesQueue
+        queue = TxMessagesQueue(max_tx_size=10 * 1024, max_tx_count=21)
+
+        for tx_param in tx_items:
+            tx_item = TxItem.create_tx_item(tx_param, 'icon_dex')
+            queue.append(tx_item)
+
+        # when get TxMessages from TxMessagesQueue
+        tx_messages = queue.get()
+        tx_total_size = tx_messages.total_size
+        tx_total_count = len(tx_messages)
+
+        print(f"messages size : {tx_total_size}")
+        print(f"messages count : {tx_total_count}")
+        print(f"queue remain : {queue.remain()}")
+
+        # then size less than max size or count less than max count
+        assert (tx_total_size < queue.max_tx_size
+                or tx_total_count < queue.max_tx_count)
+
+        # when get messages
+        messages = tx_messages.get_messages()
+
+        # then messages count equals tx_total_count
+        assert len(messages) == tx_total_count
+
+    @pytest.mark.parametrize("tx_items", [tx_count], indirect=True)
+    def test_tx_message_queue_thread_safe(self,
+                                          tx_items: List[Tuple[Transaction, TransactionVersioner]],
+                                          tx_item: Callable[[TransactionVersioner], Transaction]):
+        print()
+        # given
+        assert_count = {'produce_count': 150, 'consume_count': 0}
+
+        queue = TxMessagesQueue(max_tx_size=10 * 1024, max_tx_count=21)
+
+        tx_versioner = TransactionVersioner()
+
+        init_tx_count = len(tx_items)
+        for tx_param in tx_items:
+            init_tx_item = TxItem.create_tx_item(tx_param, 'icon_dex')
+            queue.append(init_tx_item)
+
+        # when produce new tx_item and consume tx_messages from queue
+        def producer():
+            for _ in range(assert_count.get('produce_count') - init_tx_count):
+                tx = tx_item(tx_versioner)
+                p_tx_item = TxItem.create_tx_item((tx, tx_versioner), 'icon_dex')
+                queue.append(p_tx_item)
+                print(f"producer() : queue = {queue}")
+                time.sleep(0.001)
+
+        def consumer(a_count):
+            while True:
+                tx_messages = queue.get()
+                a_count['consume_count'] += len(tx_messages)
+                message = tx_messages.get_messages()
+                print(f"consumer() : queue = {queue}, tx_messages({id(tx_messages)})")
+                print(f"consume_count : {a_count['consume_count']}")
+                if not queue.remain():
+                    break
+                time.sleep(0.3)
```

Comment:
Consider to reduce loop interval time.
If: 
- `producer` can 300 `TxItem`s per consumer loop **AND**
- `max_tx_count` of `queue` is set 21
then:
- This test **Always** get `TxMessages` from `TxMessagesQueue._queue`, not from `TxMessagesQueue._tx_messages` which might break thread safeness.